### PR TITLE
Change this "try" to a try-with-resources

### LINKED ISSUE #24: Change multiple "try" blocks to a try-with-resources
**Describe the technical debt**
Java 7 introduced the try-with-resources statement, which guarantees that the resource in question will be closed. Since the new syntax is closer to bullet-proof, it should be preferred over the older try/catch/finally version.

This rule checks that close-able resources are opened in a try-with-resources statement.

**Expected behavior**
Resource should be closed without a finally block.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by Beavan1997**
Closes #24

### CODE REVIEW

**Comment by Beavan1997** [comment]
Hi @AbhiMavani @pritesh-78 @YajingLiu2357 @luyongtang 
Created this PR for Issue #24 , Kindly review and merge the same

File: robocode.battle/src/main/java/net/sf/robocode/recording/RecordManager.java [hunk 1]
```diff
@@ -148,64 +148,58 @@ ITurnSnapshot readSnapshot() {
...
+        try (
+                FileInputStream fis = new FileInputStream(recordFilename);
+                BufferedInputStream bis = new BufferedInputStream(fis, 1024 * 1024);
+                ZipInputStream zis = (format == BattleRecordFormat.BINARY_ZIP || format == BattleRecordFormat.XML_ZIP) ? new ZipInputStream(bis) : null;
+                FileOutputStream fos = (format == BattleRecordFormat.BINARY || format == BattleRecordFormat.BINARY_ZIP || format == BattleRecordFormat.XML_ZIP || format == BattleRecordFormat.XML) ? new FileOutputStream(tempFile) : null;
+                BufferedOutputStream bos = (fos != null) ? new BufferedOutputStream(fos, 1024 * 1024) : null;
+                ObjectOutputStream oos = (bos != null && (format == BattleRecordFormat.BINARY || format == BattleRecordFormat.BINARY_ZIP)) ? new ObjectOutputStream(bos) : null
+        ) {
+            // Extracted ternary operation for ObjectInputStream initialization
+            ObjectInputStream ois = null;
```
Comment (YajingLiu2357) [inline robocode.battle/src/main/java/net/sf/robocode/recording/RecordManager.java:162]: I am not sure about this part. Maybe we can divide this part into several try blocks under specific if conditions.

ois may also need to be defined in the try-with-resource to close

**Review by YajingLiu2357 — COMMENTED** [review]
I am not sure about how to use try-with-resource. Maybe we can also wait for others' review.

File: robocode.battle/src/main/java/net/sf/robocode/recording/RecordManager.java (hunk 1 above)
Comment (Beavan1997) [reply robocode.battle/src/main/java/net/sf/robocode/recording/RecordManager.java:162]: Creating multiple if blocks will create code repetition as the conditional statement are used to use the same snippet of code.

Adding OIS in the try statement generates a new technical debt of "Nested statements" and to resolve that we would have to write it as above.

**Pull request merged**